pyi.py

```diff
@@ -396,8 +396,19 @@ def _is_type_or_Type(node: ast.expr) -> bool:
     return cls_name in {"type", "Type"}
 
 
-def _is_None(node: ast.expr) -> bool:
-    return isinstance(node, ast.Constant) and node.value is None
+if sys.version_info >= (3, 9):
+
+    def _is_None(node: ast.expr) -> bool:
+        return isinstance(node, ast.Constant) and node.value is None
+
+else:
+
+    def _is_None(node: Union[ast.expr, ast.slice]) -> bool:
+        if isinstance(node, ast.Constant):
+            return node.value is None
+        elif isinstance(node, ast.Index):
+            return node.value is None
+        return False
```

Comment:
This is to make mypy on py38 happy - there might be an easier way to do that, but couldn't think of any for now..

Reply:
It's a false positive from mypy (though mypy isn't to blame here), because mypy doesn't know that we're getting rid of all the `ast.Index` nodes using the `LegacyNormalizer`. I think all the `_SliceContents` changes you made in 0c5185f4134941f7b265d8256332daf4a239f021 are fine, but for this I think I'd maybe prefer to just `type: ignore` the problem away... you can stop mypy complaining about the "unused `type: ignore`" on Python 3.12 by doing something like `# type: ignore[arg-type,unused-ignore]`

Reply:
I can't wait until Python 3.8 goes EOL and we drop support :D